pandas/core/indexes/base.py

```diff
@@ -618,6 +618,9 @@ def where(self, cond, other=None):
         if other is None:
             other = self._na_value
 
+        if isinstance(other, ABCSparseArray):
```

Comment:
we shouldn't have Sparse specific things like this, use ``._values``

Reply:
there isn't ._values on SparseArray though? Unless I'm missing something out of @TomAugspurger 's pull request...